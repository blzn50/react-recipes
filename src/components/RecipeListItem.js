import React from 'react';
import PropTypes from 'prop-types';

const RecipeListItem = ({
  recipe, onHandleClick, onFavorited, favorited,
}) => (
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    key={recipe.id}
    onClick={() => onHandleClick(recipe.id)}
  >
    <span
      className="mr1"
      onClick={(e) => {
        e.stopPropagation();
        onFavorited(recipe.id);
      }}
    >
      {favorited ? 'âœ…' : 'ðŸ”²'}
    </span>
    <span>{recipe.name}</span>
    <span>{recipe.category}</span>
  </li>
);

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  favorited: PropTypes.bool,
  onHandleClick: PropTypes.func,
  onFavorited: PropTypes.func,
};

export default RecipeListItem;
