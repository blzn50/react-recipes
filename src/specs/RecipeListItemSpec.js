import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import RecipeListItem from '../components/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'Test recipe',
  category: 'Test category',
};

describe('<RecipeListItem />', () => {
  test('should not break if no recipe passed', () => {
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should correctly render recipe', () => {
    const component = renderer.create(<RecipeListItem recipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('should render favorited state', () => {
    const component = renderer.create(<RecipeListItem recipe={testRecipe} favorited />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should call handleClick when clicked', () => {
    const onHandleClick = jest.fn();
    const component = mount(<RecipeListItem recipe={testRecipe} onHandleClick={onHandleClick} />);
    component.simulate('click');

    expect(onHandleClick.mock.calls.length).toBe(1);
  });

  test('should call onFavorited on favorited', () => {
    const onFavorited = jest.fn();
    const component = mount(<RecipeListItem recipe={testRecipe} onFavorited={onFavorited} />);
    component
      .find('span')
      .first()
      .simulate('click');

    expect(onFavorited.mock.calls.length).toBe(1);
  });

  test('should not call onHandleClick when onFavorited is called', () => {
    const onHandleClick = jest.fn();
    const onFavorited = jest.fn();
    const component = mount(
      <RecipeListItem
        recipe={testRecipe}
        onHandleClick={onHandleClick}
        onFavorited={onFavorited}
      />,
    );
    component
      .find('span')
      .first()
      .simulate('click');

    expect(onHandleClick.mock.calls.length).toBe(0);
    expect(onFavorited.mock.calls.length).toBe(1);
  });
});
