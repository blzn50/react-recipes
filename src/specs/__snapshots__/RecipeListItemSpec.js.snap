// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> should correctly render recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    className="mr1"
    onClick={[Function]}
  >
    ðŸ”²
  </span>
  <span>
    Test recipe
  </span>
  <span>
    Test category
  </span>
</li>
`;

exports[`<RecipeListItem /> should not break if no recipe passed 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    className="mr1"
    onClick={[Function]}
  >
    ðŸ”²
  </span>
  <span />
  <span />
</li>
`;

exports[`<RecipeListItem /> should render favorited state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    className="mr1"
    onClick={[Function]}
  >
    âœ…
  </span>
  <span>
    Test recipe
  </span>
  <span>
    Test category
  </span>
</li>
`;
